packages/server/src/lib/ab.ts

```diff
@@ -56,3 +62,100 @@ export const selectVariant = <V extends Variant, T extends Test<V>>(test: T, mvt
 
     return selectWithSeed(mvtId, seed, test.variants);
 };
+
+// Amounts test code should match the equivalent code in support-frontend as closely as possible
+const amountsRandomNumber = (mvtId: number, seed: number): number => {
+    const rng = seedrandom(`${mvtId + seed}`);
+    return Math.abs(rng.int32());
+};
+// function randomNumber(mvtId: number, seed: number): number {
+//     const rng = seedrandom(`${mvtId + seed}`);
+//     return Math.abs(rng.int32());
+// }
+
+export const selectAmountsTestVariant = (
+    test: ModifiedChoiceCardAmounts,
+    countryGroupId: CountryGroupId,
+    mvtId: number,
+): SelectedAmountsVariant => {
+    const regionTest = test[countryGroupId];
+
+    const { name, variants, testIsLive, seed } = regionTest;
+
+    const seedNumber = parseInt(seed, 10);
+
+    // Safety, in case something goes wrong
+    const defaultAmounts = {
+        ONE_OFF: {
+            amounts: [1, 5, 10, 20],
+            defaultAmount: 5,
+        },
+        MONTHLY: {
+            amounts: [1, 5, 10, 20],
+            defaultAmount: 5,
+        },
+        ANNUAL: {
+            amounts: [10, 50, 100, 150],
+            defaultAmount: 50,
+        },
+    };
+
+    // The default hardcoded packet
+    const packet: SelectedAmountsVariant = {
```

Comment:
`packet` seems a strange name here - this looks like a default amounts object.
But - what happens if this gets used and carries through to the checkout page? Does support-frontend need to know about it? Would it be better to display no amounts in the epic if nothing has been configured?

Reply:
Good catch. Returning null/undefined would make more sense. I'll recode here, and further downstream to make sure that if such a situation arises then no amounts test data gets added to the CTA URL's ABtests Array

Reply:
Fixed. Also: const renamed.